### DISCUSSION TITLE
我是使用esir编译的佛跳墙固件，安装ipk后无法使用。现在也来不及编译了，node.js也无法使用

### DISCUSSION THREAD

**Discussion by akiooo45** [UnblockNeteaseMusic/luci-app-unblockneteasemusic#150] [Q&A]
root@OpenWrt:~# unm-debug
Launching luci-app-unblockneteasmusic Debugging Tool...

OpenWrt info:
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='Buddha Version'
DISTRIB_TARGET='x86/64'
DISTRIB_ARCH='x86_64'
DISTRIB_TAINTS='no-all override'
DISTRIB_REVISION='R22.4.1 Buddha V2 [2022] by eSir'
DISTRIB_DESCRIPTION='OpenWrt '

uclient-fetch info:
Package: uclient-fetch
Version: 2021-05-14-6a6011df-1
Depends: libc, libuclient20201210
Provides: wget
Status: install user installed
Section: net
Architecture: x86_64
Size: 7941
Filename: uclient-fetch_2021-05-14-6a6011df-1_x86_64.ipk
Description: Tiny wget replacement using libuclient
Installed-Time: 1651317755

Package: libustream-wolfssl20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20211120, libwolfssl5.2.0.5fb91bea
Provides: libustream-wolfssl
Conflicts: libustream-openssl
Status: unknown ok not-installed
Section: libs
Architecture: x86_64
Size: 5073
Filename: libustream-wolfssl20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (wolfssl)

Package: libustream-openssl20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20210516, libopenssl1.1
Provides: libustream-openssl
Status: install user installed
Section: libs
Architecture: x86_64
Size: 5521
Filename: libustream-openssl20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (openssl)
Installed-Time: 1651317755

Package: libustream-mbedtls20201210
Version: 2022-01-16-868fd881-1
Depends: libc, libubox20211120, libmbedtls12
Provides: libustream-mbedtls
Conflicts: libustream-openssl, libustream-wolfssl
Status: unknown ok not-installed
Section: libs
Architecture: x86_64
Size: 5338
Filename: libustream-mbedtls20201210_2022-01-16-868fd881-1_x86_64.ipk
Description: ustream SSL Library (mbedtls)

Downloading 'https://api.github.com/repos/UnblockNeteaseMusic/server/commits?sha=enhanced&path=precompiled'
Connecting to 192.30.255.117:443
Writing to stdout

Download completed (125581 bytes)
5fdfda6e606550d0dbfda12142f638235a3bd3ef

Node.js info:
Package: node
Version: v16.14.2-1
Depends: libc, libstdcpp6, libopenssl1.1, zlib, libnghttp2-14, libuv1, libcares, libatomic1
Status: install ok installed
Section: lang
Architecture: x86_64
Size: 11390938
Filename: node_v16.14.2-1_x86_64.ipk
Description: Node.js® is a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node.js' package ecosystem, npm, is the largest ecosystem of open source libraries in the world. *** The following preparations must be made on the host side. *** 1. gcc 8.3 or higher is required. 2. To build a 32-bit target, gcc-multilib, g++-multilib are required. 3. Requires libatomic package. (If necessary, install the 32-bit library at the same time.) ex) sudo apt-get install gcc-multilib g++-multilib
Installed-Time: 1652594879

Node.js is placed at /usr/bin/node
Node.js version: Not Found

luci-app-unblockneteasmusic info:
Package: luci-app-unblockneteasemusic
Version: 2.12-4
Depends: libc, busybox, dnsmasq-full, ipset, jsonfilter, node, uclient-fetch
Status: install user installed
Architecture: all
Conffiles:
 /etc/config/unblockneteasemusic e9413f9ce065cfb7b875d1948c6d3258772ab8854b7ccf193df49c3c8b6cca20
Installed-Time: 1652594879

-rw-r--r--    1 root     root         285 May 15 14:13 /etc/config/unblockneteasemusic
-rwxr-xr-x    1 root     root       10.6K Apr 25 08:21 /etc/init.d/unblockneteasemusic

/usr/share/unblockneteasemusic:
drwxr-xr-x    2 root     root        3.4K May 15 14:08 core
-rw-r--r--    1 root     root          41 May 15 14:08 core_local_ver
-rwxr-xr-x    1 root     root        5.5K Apr 25 08:21 debugging.sh
-rwxr-xr-x    1 root     root         348 Apr 25 08:21 log_check.sh
drwxr-xr-x    2 root     root        3.4K May 15 14:07 rules
-rwxr-xr-x    1 root     root        2.9K Apr 25 08:21 update.sh

config unblockneteasemusic 'config'
	option use_custom_cookie '0'
	option auto_update '1'
	option update_time '3'
	option advanced_mode '0'
	option enable '1'
	option music_source 'kuwo'
	option local_vip '1'
	option enable_flac '1'
	option replace_music_source 'lower_than_999kbps'

UnblockNeteaseMusic Node.js info:
Git HEAD version: 5fdfda6e606550d0dbfda12142f638235a3bd3ef
Core version: Not Found
-rw-r--r--    1 root     root      472.8K May 15 14:08 app.js
-rw-r--r--    1 root     root      416.2K May 15 14:08 bridge.js
-rw-r--r--    1 root     root         790 May 15 14:08 ca.crt
-rw-r--r--    1 root     root         843 May 15 14:08 server.crt
-rw-r--r--    1 root     root         359 May 15 14:08 server.key

Netease networking info:
< Location: http://m10.music.126.net/20220515145238/6db3a6c0a8d8d6c62bce49aadcc91a9b/ymusic/obj/w5zDlMODwrDDiGjCn8Ky/3057940231/3d32/93a3/4db1/0757f3a378b41e09ca2767c1abd47afa.mp3
{"code":200,"data":{"music.163.com":{"ip":["112.13.119.18","112.13.122.4","59.111.19.33","59.111.160.195","182.92.170.253","193.112.159.225","39.105.175.128","47.100.127.239","118.24.63.156","39.105.63.80"],"ipv4ScoreDelay":0,"prefer":"ipv4","score":false,"ttl":300}}}

Port status:
No instance found on port 5200.
No instance found on port 5201.

Running info:
PROCD running status: running
Thread is not found.

Firewall info:
Chain netease_cloud_music (2 references)
target     prot opt source               destination         
RETURN     all  --  anywhere             0.0.0.0/8           
RETURN     all  --  anywhere             10.0.0.0/8          
RETURN     all  --  anywhere             127.0.0.0/8         
RETURN     all  --  anywhere             169.254.0.0/16      
RETURN     all  --  anywhere             172.16.0.0/12       
RETURN     all  --  anywhere             192.168.0.0/16      
RETURN     all  --  anywhere             base-address.mcast.net/4 
RETURN     all  --  anywhere             240.0.0.0/4         
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_http src tcp dpt:www redir ports 5200
REDIRECT   tcp  --  anywhere             anywhere             ! match-set acl_neteasemusic_https src tcp dpt:https redir ports 5201

Name: neteasemusic
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0xac5177f8
Size in memory: 920
References: 2
Number of entries: 18
Members:
223.252.199.67
193.112.159.225
59.111.181.38
182.92.170.253
59.111.160.195
47.100.127.239
112.13.119.17
59.111.19.33
59.111.181.35
118.24.63.156
39.105.175.128
112.13.119.18
112.13.122.1
223.252.199.66
39.105.63.80
59.111.181.60
59.111.160.197
112.13.122.4

Name: acl_neteasemusic_http
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0x0a8abe09
Size in memory: 200
References: 1
Number of entries: 0
Members:

Name: acl_neteasemusic_https
Type: hash:ip
Revision: 5
Header: family inet hashsize 1024 maxelem 65536 bucketsize 12 initval 0x2a6aaa17
Size in memory: 200
References: 1
Number of entries: 0
Members:

dhcp-option=252,http://192.168.1.120:5200/proxy.pac
ipset=/.music.163.com/neteasemusic
ipset=/interface.music.163.com/neteasemusic
ipset=/interface3.music.163.com/neteasemusic
ipset=/apm.music.163.com/neteasemusic
ipset=/apm3.music.163.com/neteasemusic
ipset=/clientlog.music.163.com/neteasemusic
ipset=/clientlog3.music.163.com/neteasemusic

Testing source replacing...
{"data":[{"id":33190502,"url":null,"br":0,"size":0,"md5":null,"code":404,"expi":1200,"type":null,"gain":0.0,"fee":0,"uf":null,"payed":0,"flag":257,"canExtend":false,"freeTrialInfo":null,"level":null,"encodeType":null,"freeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"listenType":null},"freeTimeTrialPrivilege":{"resConsumable":false,"userConsumable":false,"type":0,"remainTime":0},"urlSource":0}],"code":200}
curl: (7) Failed to connect to 192.168.1.120 port 5200 after 0 ms: Connection refused

Log is available at:
https://litter.catbox.moe/3fswha.txt
https://transfer.sh/W5Sz9N/unm-debugging-output.txt
root@OpenWrt:~#

**Comment by ximiTech** [comment]
Status: unknown ok not-installed 缺依赖

**Comment by 1715173329** [comment]
你好像说了，又好像什么都没说。
可以用我们官方的固件： https://downloads.immortalwrt.org/releases/21.02.0/targets/x86/64/
